lib/chef/http.rb

```diff
@@ -381,7 +381,11 @@ def send_http_request(method, url, base_headers, body, &response_handler)
         elsif redirect_location = redirected_to(response)
           if [:GET, :HEAD].include?(method)
             follow_redirect do
-              send_http_request(method, url + redirect_location, headers, body, &response_handler)
+              redirected_url = url + redirect_location
+              if url.host != redirected_url.host
+                headers.delete('Authorization') if headers['Authorization']
```

Comment:
```
headers.delete('Authorization') if url.host != redirected_url.host
```
IMO, in Ruby you don't need to be that careful when removing from hashes

Reply:
Fixed.